mirrord/layer/src/lib.rs

```diff
@@ -151,6 +155,22 @@ fn layer_pre_initialization() -> Result<(), LayerError> {
         .and_then(|arg| arg.split('/').last())
         .ok_or(LayerError::NoProcessFound)?;
 
+    // This is for better SIP handling on IDEs. If started by an IDE, SIP handling was not called
+    // before starting the binary, (but we're here so binary was not SIP). This means a temp dir
+    // was not yet set, and was not added to excluded files. In that case, set it now, before
+    // generating the configuration so that if the application calls `execve`, and we patch a SIP
+    // executable, the file hooks bypass our patched files (currently there is no way to add files
+    // to the file filter after it was constructed).
+    #[cfg(target_os = "macos")]
+    let _tmp_dir = get_tmp_dir()
+        .inspect_err(|err| {
+            trace!(
+                "Getting temp dir failed with {:?} in layer_pre_initialization",
+                err
+            )
+        })
+        .unwrap_or_default();
```

Comment:
Added this part now, sorry @meowjesty  for adding stuff after the first review. But I think it's a small change that brings some value.